pysph/sph/wc/edac.py

```diff
@@ -734,6 +746,16 @@ def _get_internal_flow_equations(self):
                                     gx=self.gx, gy=self.gy, gz=self.gz),
                 SetWallVelocity(dest=solid, sources=fluids_with_io),
             ])
+        for solid in self.inviscid_solids:
+            group1.extend([
+                NoSlipVelocityExtrapolation(
```

Comment:
Since you are using these together, why not do both in the same equation, in fact, this will give you an error as `d_wij` will be added to twice, once for each equation leading to incorrect solid wall velocities.

Reply:
We need to make sure this mistake is not happening already as we use d_wij already in a few places.

Reply:
yes I will look into it.